text/1615-cargo-platform-standards.md

```diff
@@ -0,0 +1,170 @@
+- Feature Name: N/A
+- Start Date: 2016-05-14
+- RFC PR:
+  [rust-lang/rfcs#1615](https://github.com/rust-lang/rfcs/pull/1615)
+- Cargo Issues:
+  [rust-lang/cargo#1734](https://github.com/rust-lang/cargo/issues/1734)
+  [rust-lang/cargo#1976](https://github.com/rust-lang/cargo/issues/1976)
+  [rust-lang/cargo#2127](https://github.com/rust-lang/cargo/pull/2127)
+- Rustup Issues:
+  [rust-lang-nursery/rustup.rs#247](https://github.com/rust-lang-nursery/rustup.rs/issues/247)
+  [rust-lang-nursery/rustup.rs#473](https://github.com/rust-lang-nursery/rustup.rs/issues/473)
+
+# Summary
+
+Improve Cargo's integration into the host operating system by using
+platform-specific paths for config files, cache files and executables,
+so it interacts well with other tools on that platform.
+
+# Motivation
+
+Currently, Cargo puts all its files in a directory named `.cargo` below the
+home directory of the user. Using this proposal, it would reuse existing
+standard directories of the host platform. This allows other tools oblivious to
+Cargo to function in the best way.
+
+Benefits include:
+
+* Using a `.cargo` directory is violating the recommendations/rules on most
+  operating systems. _(Linux, Windows, macOS. Especially painful on Windows,
+  where dotfiles are not hidden.)_
+* Putting caches in designated cache directories allows backup tools to ignore
+  them. _(Linux, Windows, macOS. Example: Time Machine ignores the cache
+  directory on macOS.)_
+* It makes it easier for users to manage, share and version-control their
+  configuraton files, as configuration files from different applications end up
+  in the same place, instead of being intermingled with cache files. _(Linux
+  and macOS.)_
+* Cargo contributes to the slow cleanup of the `$HOME` directory by stopping to
+  add its application-private clutter to it. _(Linux.)_
+* Using a standard directory for binary outputs can allow the user to execute
+  Cargo-installed binaries without modifying their `PATH` variable. _(Linux)_
+
+Solving this problem will likely also solve the same problem in Cargo-related
+tools such as `rustup` as their strategy is "do what Cargo does".
```

Comment:
Ish. Rustup did move *away* from Appdata\Local back in 2016 to line up with Cargo, but rustup has its own .rustup directory, and will need to migrate independently - and any migration of cargo bin directories will need to be coordinated with rustup as well. I don't think it makes it easier for rustup to follow XDG for rustups own files, but it would make it easier for the files that live in `CARGO_HOME/bin`, which is effectively the only intersection point.